devicetypes/gs/gatorsystem-homewatcher.src/gatorsystem-homewatcher.groovy

```diff
@@ -0,0 +1,191 @@
+	return result
+}
+
+private Map parseReportAttributeMessage(String description) {
+	Map descMap = (description - "read attr - ").split(",").inject([:]) { map, param ->
+		def nameAndValue = param.split(":")
+		map += [(nameAndValue[0].trim()):nameAndValue[1].trim()]
+	}
+	log.debug "Desc Map: $descMap"
+
+	Map resultMap = [:]
+	if (descMap.cluster == "0500" && descMap.attrId == "0002") {
+	  def value = descMap.value.endsWith("01") ? "active" : "inactive"  //01: alarm 1; 02: alarm 2; 
+		resultMap = getMotionResult(value)
+	}
+
+	return resultMap
+}
+
+private Map parseIasMessage(String description) {
+	ZoneStatus zs = zigbee.parseZoneStatus(description)
+	Map resultMap = [:]
+
+    def seconds01 = 60
+    def seconds02 = 120
+    if (zs.isAlarm1Set()) {
+    	runIn(seconds02, stopMotion) 
+    	resultMap = getMotionResult('active')  
+    } else if (zs.isBatterySet()){
+    	resultMap = getBatteryResult(10)
+    } else if (description.contains('0x8000')) {
+	    runIn(seconds01, resetOccupancy)    
+    	resultMap = getMotionResult('occupied') 
+    } else if (description.contains('0x4000')) {    
+    	resultMap = getMotionResult('openned') 
+    } else if (description.contains('0x2000')) {    
+    	resultMap = getMotionResult('closed')  	 	
+    } else if (description.contains('0x1000')) {
+    	resultMap = getBatteryResult(100)
+    } else if (description.contains('0x0800')) {
+    	resultMap = getBatteryResult(0)
+    }  
+}
+
+private Map getBatteryResult(rawValue) {
+	log.debug "Battery rawValue = ${rawValue}"
+    sendEvent(name:"battery", value:rawValue)
+	return result
+}
+
+private Map getMotionResult(value) {
+	if (value == 'active') {
+        log.debug 'intrusion'
+        String descriptionText = "{{ device.displayName }} detected intrusion"
+        return [
+            name: 'motion',
+            value: value,
+            descriptionText: descriptionText,
+            translatable: true
+        ]
+    } else if (value == 'occupied') {
+    	log.debug 'detected occupancy'
+        String descriptionText = "{{ device.displayName }} detected occupant"
+		return [
+            name: 'presence',
+            value: "present",
+            descriptionText: descriptionText,
+            translatable: true
+        ]
+    } else if (value == 'openned') {
+    	log.debug 'detected window openned'
+        String descriptionText = "{{ device.displayName }} detected window openned"
+	    return [
+            name: 'contact',
+            value: "open",
+            descriptionText: descriptionText,
+            translatable: true
+        ]
+    } else if (value == 'closed') {
+    	log.debug 'detected  window closed'
+        String descriptionText = "{{ device.displayName }} detected window closed"
+        return [
+            name: 'contact',
+            value: "closed",
+            descriptionText: descriptionText,
+            translatable: true
+        ]
+    }
+}
+
+def installed() {
+	initialize()
+}
+def updated() {
```

Comment:
This method will be run when user changes for example device name in the application.
Are You sure You want to call initialize() ? That method will send initial events (and those might be cause of triggering automations unnecessarily - if they will be defined by the user for this device).